circuits/cpp/src/aztec3/circuits/abis/rollup/nullifier_leaf_preimage.hpp

```diff
@@ -21,7 +21,12 @@ template <typename NCT> struct NullifierLeafPreimage {
     MSGPACK_FIELDS(leaf_value, next_index, next_value);
     bool operator==(NullifierLeafPreimage<NCT> const&) const = default;
 
-    fr hash() const { return stdlib::merkle_tree::hash_multiple_native({ leaf_value, next_index, next_value }); }
+    bool is_empty() const { return leaf_value == fr(0) && next_index == fr(0) && next_value == fr(0); }
+
+    fr hash() const
+    {
+        return is_empty() ? fr(0) : stdlib::merkle_tree::hash_multiple_native({ leaf_value, next_index, next_value });
```

Comment:
silly nit: but `fr.ZERO` over `fr(0)`, here and in above function. Furthermore i think fr has a `.is_zero()` method which is more erigonomic